objects/television.rb

```diff
@@ -0,0 +1,122 @@
+#! /usr/bin/env ruby
+
+=begin
+
+Create a new Television with 'new'.  It has channels 3 to 500 and the volume goes from
+ 0 to 25.
+
+ - turn it on or off by calling 'power'; nothing else works unless it's on
+ - 'channel_up'/'channel_down' will go up or down to the next channel; it will loop when
+   reaching the first or last channel.
+ - 'channel_set(999)' will jump to the selected channel as long as it's within 3-500.
+ - 'channel_swap' will change to whatever previous channel you were on.
+ - 'volume_up'/'volume_down' will increase or decrease the volume by 1 until the upper
+   or lower limit is reached.
+ - 'mute' sets the volume to 0; if the volume is already zero, it sets it to whatever it
+   was before mute was called.
+ - 'status' displays the current channel and volume settings.
+
+=end
+
+class Television
+  attr_reader :on
+  alias_method :on?, :on
+
+  CHANNEL_RANGE = 3..500
+  VOLUME_RANGE = 0..25
+
+  def initialize
+    @on = false
+    @channel = 3
+    @prev_channel = 3
+    @volume = 0
+    @pre_mute_volume = 0
+  end
+
+  def power
+    @on = !@on
+    puts "TV is now #{on? ? 'on' : 'off'}"
+  end
+
+  def set_channel(number)
+    if on?
```

Comment:
You may want to change this to a logic gate...

    def set_channel(number)
      return unless on?
      ...
   end